sdk/cosmosdb/cosmos/src/client/ChangeFeed/ChangeFeedIteratorBuilderClass.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import type { ChangeFeedIteratorResponse } from "./ChangeFeedIteratorResponse";
+import type { Container, Resource } from "../../client";
+import type { ClientContext } from "../../ClientContext";
+import type { ChangeFeedPullModelIterator } from "./ChangeFeedPullModelIterator";
+import type { ChangeFeedIteratorOptions } from "./ChangeFeedIteratorOptions";
+import { changeFeedIteratorBuilder } from "./changeFeedIteratorBuilder";
+import type { PartitionKeyRangeCache } from "../../routing";
+
+/**
+ * @hidden
+ * Provides iterator for change feed.
+ *
+ * Use `Items.getChangeFeedIterator()` to get an instance of the iterator.
+ */
+
+export class ChangeFeedIteratorBuilder<T> implements ChangeFeedPullModelIterator<T> {
+  private iterator: ChangeFeedPullModelIterator<T>;
+  private isInstantiated: boolean;
+  /**
+   * @internal
+   */
+  constructor(
+    private cfOptions: ChangeFeedIteratorOptions,
+    private clientContext: ClientContext,
+    private container: Container,
+    private partitionKeyRangeCache: PartitionKeyRangeCache,
+  ) {
+    this.isInstantiated = false;
+  }
+
+  /**
+   * Change feed is an infinite feed. hasMoreResults is always true.
+   */
+  get hasMoreResults(): boolean {
+    return true;
+  }
+
+  /**
+   * Gets an async iterator which will yield change feed results.
+   */
+  public async *getAsyncIterator(): AsyncIterable<ChangeFeedIteratorResponse<Array<T & Resource>>> {
+    if (!this.isInstantiated) {
+      await this.instantiateIterator();
+    }
+    do {
+      const result = await this.iterator.readNext();
+      yield result;
+    } while (this.hasMoreResults);
```

Comment:
would this be always true?

Reply:
Yes, Change feed is an infinite feed. hasMoreResults is always true.